python/Cargo.toml

```diff
@@ -30,16 +30,16 @@ edition = "2018"
 libc = "0.2"
 tokio = { version = "1.0", features = ["macros", "rt", "rt-multi-thread", "sync"] }
 rand = "0.7"
-pyo3 = { version = "0.14.1", features = ["extension-module"] }
-datafusion = { git = "https://github.com/apache/arrow-datafusion.git", rev = "4d61196dee8526998aee7e7bb10ea88422e5f9e1" }
+pyo3 = { version = "0.14.2", features = ["extension-module"] }
+datafusion = { path = "../datafusion", features = ["pyarrow"] }
 
```

Comment:
same here, this you should avoid a path-only dependency here, but it looks like there might be other considerations I'm unaware of given the use of git.

Reply:
I updated both arrow-rs and datafusion which are dependencies of this change, this is a quazy "editable mode".

Reply:
Using path here is fine, we do the same thing for ballista as well. Ideally, if the crate needs to be published on crates.io, we will use both `path` and `version` so it will work with `cargo publish`. But in this case, version is not required since we don't have plan to publish the python binding crate to crates.io.